momepy/coins.py

```diff
@@ -0,0 +1,998 @@
+<<<<<<< HEAD:PythonTool/200330_NetworkContinuityMultiprocessing_PT_Rework_Final.py
+"""
+This Python script generates strokes from the line type ESRI shapefiles, mainly roads.
+
+Author: Pratyush Tripathy
+Date: 04 January 2020
+Version: 0.3
+
+The script is a supplementary material to the full length article:
+
+Title:
+An open-source tool to extract natural continuity and hierarchy of urban street networks
+
+Journal:
+Environment and Planning B: Urban Analytics and City Science
+
+Authors:
+Pratyush Tripathy, Pooja Rao, Krishnachandran Balakrishnan, Teja Malladi
+
+GitHub repository:
+https://github.com/PratyushTripathy/NetworkContinuity
+
+"""
+
+########################################################################################
+########################################################################################
+#################   PLEASE DO NOT EDIT THE BELOW PART OF THE CODE    ###################
+#####   SCROLL DOWN TO THE LOWER EXTREME OF THE SCRIPT TO CHANGE INPUT FILE NAME   #####
+########################################################################################
+########################################################################################
+
+import os, sys, math, time, multiprocessing
+from functools import partial
+import numpy as np
+import shapefile as shp
+
+#Set recurrsion depth limit to avoid error at a later stage
+sys.setrecursionlimit(10000)
```

Comment:
The search for adjacent street segments uses a recursion function. A lower recursion limit throws an error. But yeah, we can try to write the code differently to avoid this.